lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -943,6 +944,22 @@ bool OverrideMatcher::checkOverride(ValueDecl *baseDecl,
     emittedMatchError = true;
   }
 
+  auto baseGenericCtx = baseDecl->getAsGenericContext();
+  auto declGenericCtx = decl->getAsGenericContext();
+
+  // If the generic signatures are different, then complain.
+  if (baseGenericCtx && declGenericCtx) {
+    auto baseGenericSig = baseGenericCtx->getGenericSignature();
+    auto declGenericSig = declGenericCtx->getGenericSignature();
+    
+    if (baseGenericSig != declGenericSig) {
```

Comment:
baseGenericSig should not have any requirements not satisfied by declGenericSig, or vice versa.

Now it's only possible to build a substitution map in one direction (turning base class requirements into derived class requirements). So requirementsNotSatisfiedBy will need to take _two_ substitution maps; the first one is applied to the base requirement, the second one is applied to the derived requirement.

Your two calls both pass one empty and one non-empty substitution map, flipped around.

Reply:
Can I just call `baseGenericSig->requirementsNotSatisfiedBy(declGenericSig, subMap)` & declGenericSig->requirementsNotSatisfiedBy(baseGenericSig, subMap)` instead? For `subMap`, I can just flip it around by using `baseClass` instead of `derivedClass`.

Reply:
Hmm that seems to work on tests, but fails when building the stdlib.

Reply:
How can two maps be used inside `requirementsNotSatisfiedBy()`? Because it uses `otherSig` and calls `isRequirementSatisfied` on it and the `req` passed is from the generic signature you call `requirementsNotSatisfiedBy()` on.

Reply:
@slavapestov Could you clarify what you meant when you said "So requirementsNotSatisfiedBy will need to take two substitution maps"? When I call `requirementsNotSatisfiedBy()`, it's easy to apply a substitution to the signature I pass to it. To apply a substitution to the requirements of the other signature, I would need to extract its requirements, substitute them and then re-create a new signature (maybe skip adding the requirement if it can't be substituted?). 

Is that what you meant? Thank you!